### ISSUE TITLE
Channel doesn't hop, stuck on -1

### ISSUE THREAD

**Issue by sasquatchdev** [kriswebdev/android_aircrack#4]
This used to be a problem with an old version but I'm not sure where the problem is exactly. Everything else is working quite nicely. Is it possible to backport some code from the latest svn sources to fix this?

I've tried ath9k_htc and rt2800usb and the problem persists with both devices.

**Comment by kriswebdev** [comment]
Have you tried to use the latests Aircrack-ng (1.2-rc4) from a Linux PC and does it work?

To upgrade Aicrack-ng for Android to the latesst official Aircrack-ng version, you need to get the latests official Aircrack-ng source code (1.2-rc4), cherry-pick [this single commit](https://github.com/kriswebdev/android_aircrack/commit/49ae7e2ea2938a5f7cd349a80f0fc8e85d6afe02), fix eventual merge conflicts, and build as per Aircrack-ng for Android instructions. I'll try to find some time to do it, I need to recreate the build environment.

**Comment by sasquatchdev** [comment]
> Have you tried to use the latests Aircrack-ng (1.2-rc4) from a Linux PC and does it work?

Yes, the latest version does work. This version works just as well on x86_64 and on my Raspberry Pi, however the Android system is where I'm having problems with channel hops. I suspect the problem might be outside the scope of this project, but I'll upgrade to latest svn and check.

I'll submit a pull request if I get it working. I have a build environment ready and can provide you SSH access if you'd like to collaborate.
